### ISSUE TITLE
Allow index WP_Query override (ie. apply_filters on WP_Query before it is processed)

### ISSUE THREAD

**Issue by virtuman** [algolia/algoliasearch-wordpress#158]
### What did you expect to happen?
Would be great to have an ability to override WP_Query before it is processed during indexing. Would be even better if this new filter was "operation-aware" to distinguish between full re-index on individual product update.

### What happened instead?
Currently it indexes ALL published posts and no way to filter out posts based on some conditions. Good example would be - DO NOT INDEX: WooCommerce outofstock products.

### Technical info
* WordPress version: 4.5.3
* Algolia Search plugin version: 0.2.8

**Comment by rayrutjes** [comment]
Hey again @virtuman ;)
Again, thank you for all this feedback, much appreciated!

Part of the answer is in #159 

Regarding WooCommerce, we will release a plugin based on this once its architecture is stable. 

We still have some dead ends we need to address:
- Ordering (which requires us to support the 'slaves' feature of Algolia)
- Easier instantsearch.js integration (which requires more discussion about the best way to ease the frontend integration; shortcode? page template?; etc...)

Regarding your specific point, about 'operation-aware' the thing is that all our operations should be idempotent to ensure good consistency. I'm not sure adding the operation context would add anything to the current flow. Maybe I missed something about your use case here?

**Comment by rayrutjes** [comment]
Hi @virtuman, I haven't heard back from you, do you have any reaction on this?

**Comment by rayrutjes** [comment]
This issue makes a more sense to me than the one I just closed: #159 

Here we are talking about optimizing the the indexing part.

The thing is that if we actually allow people to pre-filter the query, then if other plugins will have a hard time figuring out why some post are not indexed.

Also, as explained in #159 we now `supress_filters` to avoid plugins such as WPML to skip some posts based on the current language, making your issue un-resovable.

Closing this for now, but again feel to open a new issue if you have other improvement ideas, very much appreciated!
